entrypoint.sh

```diff
@@ -52,7 +52,7 @@ aws() {
 }
 
 azure() {
-    test_vars AZURE_CLIENT_ID AZURE_CLIENT_SECRET AZURE_TENANT_ID AZURE_SUBSCRIPTION_ID AZURE_RESOURCE_GROUP AZURE_SUBNET_ID AZURE_IMAGE_ID
+    test_vars AZURE_CLIENT_ID AZURE_TENANT_ID AZURE_SUBSCRIPTION_ID AZURE_RESOURCE_GROUP AZURE_SUBNET_ID AZURE_IMAGE_ID
```

Comment:
tenant_id is needed?
also consider using one_of instead of test_vars , IIUC it suits here (otherwise we can actually remove it)

Reply:
Yes tenant_id is needed and in service principal case user populates it, in workload identity case it is automatically populated so yes it is needed.

Reply:
IIUC one_of is for when it is an either / or condition right? We need all the others to be set not an either or condition. But I can think of situations when we need one_of especially with the networking stuff.

Reply:
in the workload identity case AZURE_TENANT_ID is automatically populated as environment variable?

Reply:
Yep it is automatically populated.